core/src/test/java/org/wildfly/extras/sunstone/api/impl/docker/DockerNodeTest.java

```diff
@@ -535,6 +538,23 @@ public void testInsufficientCapabilities() {
         }
     }
 
+    @Test
+    public void testVolumeBindings() {
+        final int nonce = new Random().nextInt();
+        try (DockerNode node = (DockerNode) dockerProvider.createNode("alpineVolumes",
+                ImmutableMap.<String, String> of("template", "alpine", "docker.binds", "/tmp:/data"))) {
+            final ExecResult execResult = node.exec("sh", "-c", "echo " + nonce + " > /data/testVolumeBindings");
+            assertEquals(0, execResult.getExitCode());
+        }
+        try (DockerNode node = (DockerNode) dockerProvider.createNode("alpineVolumes",
+                ImmutableMap.<String, String> of("template", "alpine", "docker.binds", "/tmp:/mnt"))) {
+            final ExecResult execResult = node.exec("cat","/mnt/testVolumeBindings");
+            assertEquals(0, execResult.getExitCode());
+            assertThat(execResult.getOutput(), containsString(String.valueOf(nonce)));
+            node.exec("rm -f /mnt/testVolumeBindings");
```

Comment:
I don't think this actually works - I think `exec` quotes it in such a way that the whole command with options will be considered to be a command name and the execution will fail because such command (`'rm -f /mnt/testVolumeBindings'`) doesn't exist. Still, it doesn't matter since this is the end of the test and the node will die.

I'm not sure though, the `DockerNode` implementation is specific and quick look through JClouds Labs has shown that the implementation in there is immensely complicated (as usual).